frontend/packages/employee-frontend/src/components/unit/unit-details/UnitEditor.tsx

```diff
@@ -687,27 +692,94 @@ export default function UnitEditor(props: Props): JSX.Element {
       <FormPart>
         <div>{showRequired(i18n.unitEditor.label.canApply)}</div>
         <FixedSpaceColumn>
-          <Checkbox
-            disabled={!props.editable}
-            label={i18n.unitEditor.field.canApplyDaycare}
-            checked={form.canApplyDaycare}
-            onChange={(canApplyDaycare) => updateForm({ canApplyDaycare })}
-            dataQa="application-type-checkbox-DAYCARE"
-          />
-          <Checkbox
-            disabled={!props.editable}
-            label={i18n.unitEditor.field.canApplyPreschool}
-            checked={form.canApplyPreschool}
-            onChange={(canApplyPreschool) => updateForm({ canApplyPreschool })}
-            dataQa="application-type-checkbox-PRESCHOOL"
-          />
-          <Checkbox
-            disabled={!props.editable}
-            label={i18n.unitEditor.field.canApplyClub}
-            checked={form.canApplyClub}
-            onChange={(canApplyClub) => updateForm({ canApplyClub })}
-            dataQa="application-type-checkbox-CLUB"
-          />
+          {['daycare', 'preschool', 'club'].map((type) => {
+            const key = `${type}ApplyPeriod`
+            const period = form[key] as OpenEndedPeriod | null
```

Comment:
This unchecked cast and the other one a couple of lines later will almost certainly cause a bug later and debugging+fixing it will take more time than implementing this correctly in the first place.

It's a bit more code, but actually spelling out the different names avoids "dynamic name generation" and loss of type safety. For example:

```
[
  {type: 'daycare', checkboxI18n: 'canApplyDaycare', field: 'daycareApplyPeriod'},
  ...
].map(...)
```